std/haxe/Callback.hx

```diff
@@ -0,0 +1,62 @@
+package haxe;
+
+typedef CallbackHandler<T> = (error:Null<Error>, result:T) -> Void;
+
+/**
+	A callback.
+
+	All instances of `Callback` are one-time functions. That is, invoking a callback
+	the second time is prohibited and will produce a null pointer exception.
+
+	All callbacks in the standard library are functions which accept
+	two arguments: an error (`haxe.Error`) and a result (`T`).
+
+	Non-null `error` means an operation failed to finish successfully.
+	In case of failure the value of the second argument has no meaning and should
+	not be used.
+
+	The underlying function type type is declared in `haxe.CallbackHandler`.
+**/
+abstract Callback<T>(CallbackHandler<T>) from CallbackHandler<T> {
```

Comment:
That seems quite specific to package implementation, I'm not sure I would make it part of haxe package. Also, you might want a notImplemented callback instead of doing .fail(new NotImplemented()) for every function missing implementation ;)

Reply:
I expect it to be useful for `hxnodejs` and maybe other things once we implement coroutines.